gdal/ogr/ogrsf_frmts/shape/shpopen.c

```diff
@@ -504,21 +497,21 @@ SHPOpen( const char * pszLayer, const char * pszAccess )
 /*                                                                      */
 /*      Open the .shp and .shx files based on the basename of the       */
 /*      files or either file name.                                      */
+/*      There are 2 types of SHPOpen() : with .shx restoring code       */
+/*      (SHPOpenLLEx) and normal one.                                   */
```

Comment:
We should indeed avoid changing SHPOpenLL signature and default behaviour. My above suggestion was to have SHPOpenLLEx(existing arguments, int bCreateShxIfNotNecessary) and SHPOpenLL() call it with just bCreateShxIfNotNecessary = FALSE. That said, I didn't study deeply how much is shared between both, so perhaps have common function(s) called by both SHPOpenLL() and SHPOpenLLEx(). We should just avoid copy&paste.